odalic/src/main/java/cz/cuni/mff/xrg/odalic/outputs/rdfexport/DefaultAnnotatedTableToRDFExportAdapter.java

```diff
@@ -0,0 +1,136 @@
+    Model model = new LinkedHashModel();
+    
+    HashMap<String, Integer> columnsNamesPositionsInput = new HashMap<>();
+    int i = 0;
+    for (String headerName : extendedInput.headers()) {
+      columnsNamesPositionsInput.put(headerName, i);
+      i++;
+    }
+    
+    HashMap<String, TableColumn> namesColumnsAnnotatedTable = new HashMap<>();
+    for (TableColumn column : annotatedTable.getTableSchema().getColumns()) {
+      namesColumnsAnnotatedTable.put(column.getName(), column);
+    }
+    
+    // fetch the relations in appropriate order
+    HashMap<String, HashMap<String, HashMap<String, String>>> relations = new HashMap<>();
+    for (TableColumn column : annotatedTable.getTableSchema().getColumns()) {
+      if ((column.getSuppressOutput() == null || !column.getSuppressOutput()) && 
+          StringUtils.isNoneBlank(column.getAboutUrl(), column.getPropertyUrl())) {
+        relations.putIfAbsent(column.getAboutUrl(), new HashMap<>());
+        relations.get(column.getAboutUrl()).putIfAbsent(column.getPropertyUrl(), new HashMap<>());
+        if (column.getValueUrl() == null) {
+          relations.get(column.getAboutUrl()).get(column.getPropertyUrl()).put(String.format("{%s}", column.getName()), 
+              column.getDataType());
+        }
+        else if (isColumnLink(column.getValueUrl())) {
+          relations.get(column.getAboutUrl()).get(column.getPropertyUrl()).put(column.getValueUrl(), 
+              namesColumnsAnnotatedTable.get(column.getValueUrl().substring(1, column.getValueUrl().length()-1)).getDataType());
+        }
+        else {
+          relations.get(column.getAboutUrl()).get(column.getPropertyUrl()).put(column.getValueUrl(), null);
+        }
+      }
+    }
+    
+    // process the rows
+    IRI subject, predicate;
+    Value object;
+    for (List<String> row : extendedInput.rows()) {
+      for (Entry<String, HashMap<String, HashMap<String, String>>> subjectEntry : relations.entrySet()) {
+        for (Entry<String, HashMap<String, String>> predicateEntry : subjectEntry.getValue().entrySet()) {
+          for (Entry<String, String> objectEntry : predicateEntry.getValue().entrySet()) {
+            
+            // create the subject
+            if (isColumnLink(subjectEntry.getKey())) {
+              subject = factory.createIRI(row.get(
+                  columnsNamesPositionsInput.get(subjectEntry.getKey().substring(1, subjectEntry.getKey().length()-1))));
+            }
+            else {
+              subject = factory.createIRI(subjectEntry.getKey());
+            }
+            
+            // create the predicate
+            switch (predicateEntry.getKey()) {
+              case DCTERMS_TITLE:
+                predicate = DCTERMS.TITLE;
+                break;
+              case RDF_TYPE:
+                predicate = RDF.TYPE;
+                break;
+              default:
+                predicate = factory.createIRI(predicateEntry.getKey());
```

Comment:
You are creating the same predicate IRI for each row, but the IRI is the same for every row! So this part should be outside of foreach (Row r: rows)